### DISCUSSION TITLE
How to use terragrunt v0.21.5 and terraform v0.11.15 scripts to work with aws eu-central-2 region

### DISCUSSION THREAD

**Discussion by shubhamkj** [gruntwork-io/knowledge-base#814] [Help]
Hi ,

We have infrastructure on AWS regions us-east-1 & eu-west-1 which is using terragrunt v0.21.5 and terraform v0.11.5. And scripts are working fine. But recently we are trying to use same scripts with same terragrunt and terraform version in eu-central-2 Zurich region on aws. We are struggling to initiate scripts and getting below kind of errors -
**### Error -**
```
Create S3 bucket abc-tfstate-eu-central-2 returned an error: InvalidToken: The provided token is malformed or otherwise invalid.
	status code: 400, request id: ABCDE0PNF49E7E0K4, host id: 5dinGMCosuQt/WnZ7JeyuP4HLilCRmKrVnBevRxpuTysXg6cNw8xMNB+BmFIe4+hB/roXXPr7ko=. Sleeping for 10s and will try again.
```
I have tried few options like skip_region_validation = true in aws provider, terrafrom backend, skip_credentials_validation in remote_state block but it's not working. 
We are using aws provider plugin v2.70.4 .

Please help to fix the existing scripts, as I can see probable option is to upgrade terragrunt and terraform to latest or 1.x version.

Let me know your thoughts on this.

Thanks in adav 

---

<ins datetime="2024-04-09T11:57:36Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110869">Tracked in ticket #110869</a></p>
</ins>

**Comment by lev-ok** [comment]
As you can see from the error, the issue is with your AWS credentials.

**Reply by shubhamkj** [reply]
But AWS credentials is setup properly and I am able to execute aws sts get-caller-identity which provides me correct output of assumed IAM Role. And same is working for other region like us-east-1, eu-west-1 as well. Doesn't seem to be issue with AWS credentials here.

**Comment by denis256** [comment]
I suspect `v2.70.4` doesn't have support for this region

Looks like `eu-central-2` was introduced in since v4.40.0
https://github.com/hashicorp/terraform-provider-aws/pull/27814

**Reply by shubhamkj** [reply]
Yes, I am also suspecting this. But how to overcome is one of the challenge here. 

I am somehow able to fix it with setting AWS Region by exporting 2 variables and execute aws sts assume-role --role-arn $ASSUME_ROLE_ARN explicitly . And not using iam_role in terragrunt files.

AWS_REGION and AWS_DEFAULT_REGION currently it's working with CLI.

Trying to setup same on Jenkins pipelines.
